RockWeb/Blocks/Reporting/DataViewDetail.ascx.cs

```diff
@@ -142,6 +142,47 @@ protected void btnEdit_Click( object sender, EventArgs e )
         }
 
         /// <summary>
+        /// Handles the Click event of the Copy button control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs" /> instance containing the event data.</param>
+        protected void btnCopy_Click( object sender, EventArgs e )
+        {
+            // Create a new Data View using the current item as a template.
+            var id = int.Parse( hfDataViewId.Value );
+            
+            var service = new DataViewService( new RockContext() );
```

Comment:
@MrUpsideDown Sorry, I've been out for a few days.  I agree with not nesting too many creates in a row.  I generally create the RockContext separately and instantiate services on the fly when needed.  A lot of the service methods take an optional RockContext so you can pass in the already-created context for a (tiny) performance benefit.